Framework/Core/include/Framework/ASoA.h

```diff
@@ -1107,7 +1057,7 @@ class TableMetadata
   static const o2::framework::expressions::BindingNode _Getter_##Id { _Label_,     \
                                                                       o2::framework::expressions::selectArrowType<_Type_>() }
 
-#define DECLARE_SOA_INDEX_COLUMN(_Name_, _Getter_) DECLARE_SOA_INDEX_COLUMN_FULL(_Name_, _Getter_, int32_t, _Name_##s, "f" #_Name_ "sID")
```

Comment:
By default arrow can store 32bit values max. Why do you need this change?

Reply:
Before I completely rework the type handling, I can only infer type from table.num_rows() which returns int64_t. Actual size of index is only known at runtime, unfortunately. Alternatively I can just set everything to int32_t for now...